test/test_infectionsrtfeedback.py

```diff
@@ -39,32 +40,53 @@ def _infection_w_feedback_alt(
     tuple
     """
 
-    Rt = np.array(Rt)  # coerce from jax to use numpy-like operations
     T = len(Rt)
+    Rt = np.array(Rt).reshape(
+        T, -1
+    )  # coerce from jax to use numpy-like operations
     len_gen = len(gen_int)
-    I_vec = np.concatenate([I0, np.zeros(T)])
-    Rt_adj = np.zeros(T)
-
-    for t in range(T):
-        Rt_adj[t] = Rt[t] * np.exp(
-            inf_feedback_strength[t]
-            * np.dot(I_vec[t : t + len_gen], np.flip(inf_feedback_pmf))
-        )
-
-        I_vec[t + len_gen] = Rt_adj[t] * np.dot(
-            I_vec[t : t + len_gen], np.flip(gen_int)
-        )
-
-    return {"post_initialization_infections": I_vec[I0.size :], "rt": Rt_adj}
-
-
-def test_infectionsrtfeedback():
+    I_vec = np.concatenate([I0.reshape(T, -1), np.zeros(Rt.shape)])
+    Rt_adj = np.zeros(Rt.shape)
+    inf_feedback_strength = np.array(inf_feedback_strength).reshape(T, -1)
+
+    for n in range(Rt.shape[1]):
+        for t in range(Rt.shape[0]):
+            Rt_adj[t, n] = Rt[t, n] * np.exp(
+                inf_feedback_strength[t, n]
+                * np.dot(I_vec[t : t + len_gen, n], np.flip(inf_feedback_pmf))
+            )
+
+            I_vec[t + len_gen, n] = Rt_adj[t, n] * np.dot(
+                I_vec[t : t + len_gen, n], np.flip(gen_int)
+            )
+
+    return {
+        "post_initialization_infections": np.squeeze(I_vec[I0.shape[0] :]),
```

Comment:
Can we just construct the `I_vec[I0.shape[0] :]` portion without padding `I_vec` and then overwriting it?

Reply:
I can't think of a straightforward way of doing this since we use sliding window of values from a dynamically constructed array $[I_{- \text{init time}} ,\cdots,I_1,I_2,\cdots]$. Let me know if you have ideas.
Also, given this is meant to be a 'primitive' method to test values computed using a more efficient method, it may not be worth spending a lot of time here?

Reply:
Good point about this being part of tests. I didn't think of that when I was reviewing.